web/src/browser_history.js

```diff
@@ -19,6 +21,32 @@ export const state = {
         : `#${user_settings.default_view}`,
 };
 
+export const hash_to_page_location = new Map();
```

Comment:
I was wondering if there's a specific reason behind using a Map object for `hash_to_page_location`?

Reply:
I tend to use `Map` for when I want to map keys (e.g. hash) to values (e.g. page location) -- I'd love to hear if there is another data structure you think could work better :)

Reply:
One thing about Map is that it allows you to use non-string keys. However, in our case, it seems like we don't actually need them, so we can just go with a regular object instead, right?

If we do decide to go with Map, I it is recommended leveraging its native methods like `get` and `set` for optimal functionality.

Reply:
Ah, that's helpful information thanks! Yeah we can use a regular object then :)